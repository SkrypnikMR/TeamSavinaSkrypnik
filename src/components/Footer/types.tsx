import React from 'react';

import { ITheme } from '../App/types';

export interface IDiv extends ITheme{
    color: string;
    padding: string;
    textAlign: string;
    textDecoration: string;
    cursor: string;
}
