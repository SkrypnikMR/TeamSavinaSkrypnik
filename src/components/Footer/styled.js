import styled from "styled-components";

export const StFooter = styled.div`
    color: white;
    padding: 0 10px 10px;
    text-align: end;
    a{
        text-decoration: none;
        color: white;
        cursor: pointer;
        :hover{
            color: #23EFFC;
        }
    }
`;