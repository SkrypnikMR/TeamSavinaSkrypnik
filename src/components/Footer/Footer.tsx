import React from 'react';

import { useTheme } from '../Hook/useTheme';

import { StFooter } from './styled';

const Footer = () => {
    const { colors, theme } = useTheme();

    return (
        <StFooter colors={colors} theme={theme}>
            create for &nbsp; 
            <a href="https://deveducation.com/" target="_blank" rel="noreferrer">
                dev
            {'{Education}'} 
</a> 
        </StFooter>
    );
};

export default Footer;
