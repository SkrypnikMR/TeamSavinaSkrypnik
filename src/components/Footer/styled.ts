import styled from 'styled-components';

import { IDiv } from './types';

export const StFooter = styled.div < IDiv > `
    color: white;
    padding: 0 10px 10px;
    text-align: end;
    a{
        text-decoration: none;
        color: white;
        cursor: pointer;
        :hover{
            color: ${({ colors, theme }) => colors[theme].buttonText};
        }
    }
`;
