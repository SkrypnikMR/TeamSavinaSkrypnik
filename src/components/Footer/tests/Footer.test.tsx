import React from 'react';
import { shallow, mount } from 'enzyme';

import Footer from '../Footer';

describe('Footer', () => {
    it('Should match snapshot', () => {
        const component = shallow(<Footer />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StFooter', () => {
        const component = mount(<Footer/>);
        expect(component.find('styled__StFooter')).toHaveLength(1);
    });
    it('should render a', () => {
        const component = mount(<Footer/>);
        expect(component.find('a')).toHaveLength(1);
    });
});
