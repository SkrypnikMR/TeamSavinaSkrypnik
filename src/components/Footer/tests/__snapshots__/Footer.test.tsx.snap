// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should match snapshot 1`] = `"<div class=\\"styled__StFooter-sc-1xbiugi-0 bClKzk\\">create_forÂ <a href=\\"https://deveducation.com/\\" target=\\"_blank\\" rel=\\"noreferrer\\">dev{Education}</a></div>"`;
